### PR TITLE
Cleanup unused Azure client interfaces

### PR DESCRIPTION
**Pull request by feiskyer**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:

/kind cleanup
/area provider/azure
/sig cloud-provider

**What this PR does / why we need it**:

Cleanup unused Azure client interfaces and switch to new client interfaces in all places.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss.go [hunk 1]
```diff
@@ -40,6 +40,9 @@ import (
 )
 
 var (
+	// virtualMachineScaleSetsDeallocating indicates VMSS instances are in Deallocating state.
+	virtualMachineScaleSetsDeallocating = "Deallocating"
```
Comment (andyzhangx) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss.go]: why adding this const, it's not used in current code

Comment (feiskyer) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss.go]: it's moving from azure_client.go to azure_vmss.go, and it is used in `ensureVMSSInPool()`

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Pull request merged**